src/parser/priest/shadow/modules/features/DarkThoughts.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+
+import Analyzer, { SELECTED_PLAYER, Options } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import SPELLS from 'common/SPELLS/index';
+import SpellLink from 'common/SpellLink';
+import Statistic from 'interface/statistics/Statistic';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+import { When, ThresholdStyle } from 'parser/core/ParseResults';
+import { t } from '@lingui/macro';
+import { formatPercentage } from 'common/format';
+import EventHistory from 'parser/shared/modules/EventHistory';
+
+// Example log: /report/JAPL1zpDfN7W8wck/33-Heroic+The+Council+of+Blood+-+Kill+(5:46)/Mayrim/standard/statistics
+class DarkThoughts extends Analyzer {
+  static dependencies = {
+    eventHistory: EventHistory,
+  };
+  protected eventHistory!: EventHistory;
+
+  procsGained: number = 0;
+  procsUsed: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.DARK_THOUGHT_BUFF), this.onBuffApplied);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.DARK_THOUGHT_BUFF), this.onBuffRemoved);
+  }
+
+  onBuffApplied() {
+    this.procsGained += 1; // Add a proc to the counter
+  }
+
+  onBuffRemoved() {
+    if (!this.eventHistory.last(1, 100, Events.cast.by(SELECTED_PLAYER).spell(SPELLS.MIND_BLAST))) { // If MB is not instant, it's not a proc
+      return;
+    }
+
+    this.procsUsed += 1;
+  }
+
+  getProcsWasted() {
```

Comment:
i assume this should be 

get procsWasted() {
    return this.procsGained - this.procsUsed;
}

and then this.procsWasted / this.procsGained in actual under the suggestion threshold